From db21b5289832a461ef9ee9d27666ae959076cff8 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Wed, 6 Feb 2019 13:17:57 +0000
Subject: [PATCH 0946/1077] MANTIS 0024715 Build: VC10; fix compiler warnings

warning C4005: '_ASSERTE' : macro redefinition
warning C4715: '`anonymous namespace'::main_2nd_level_catchSE' : not all control paths return a value
warning C4723: potential divide by 0


git-svn-id: svn://localhost/SelfServiceCommon/trunk@1434 90b65887-3827-0410-9a23-83215b262276
---
 SelfServiceCommon/Massai/cpp/Systools/src/mSysDebugW32.cpp    | 4 ++++
 SelfServiceCommon/Massai/cpp/Systools/src/mSysMain.cpp        | 1 +
 .../Massai/cpp/Systools/src/testcaseException.cpp             | 2 ++
 3 files changed, 7 insertions(+)

diff --git a/SelfServiceCommon/Massai/cpp/Systools/src/mSysDebugW32.cpp b/SelfServiceCommon/Massai/cpp/Systools/src/mSysDebugW32.cpp
index ab785982..fb27c4b9 100644
--- a/SelfServiceCommon/Massai/cpp/Systools/src/mSysDebugW32.cpp
+++ b/SelfServiceCommon/Massai/cpp/Systools/src/mSysDebugW32.cpp
@@ -27,6 +27,10 @@
 using namespace std;
 #endif
 
+#ifdef _ASSERTE
+#undef _ASSERTE
+#endif
+
 #define _ASSERTE( x )
 #define VERIFY(x) (x)
 
diff --git a/SelfServiceCommon/Massai/cpp/Systools/src/mSysMain.cpp b/SelfServiceCommon/Massai/cpp/Systools/src/mSysMain.cpp
index 2996817e..7f59cb9f 100644
--- a/SelfServiceCommon/Massai/cpp/Systools/src/mSysMain.cpp
+++ b/SelfServiceCommon/Massai/cpp/Systools/src/mSysMain.cpp
@@ -92,6 +92,7 @@ namespace
          {
             throw *SysException;
          }
+         return 1;
       }
    }
 
diff --git a/SelfServiceCommon/Massai/cpp/Systools/src/testcaseException.cpp b/SelfServiceCommon/Massai/cpp/Systools/src/testcaseException.cpp
index 5d53cb3e..f9cc3727 100644
--- a/SelfServiceCommon/Massai/cpp/Systools/src/testcaseException.cpp
+++ b/SelfServiceCommon/Massai/cpp/Systools/src/testcaseException.cpp
@@ -16,6 +16,8 @@
 #include <iostream>
 using namespace std;
 
+// warning C4723: potential divide by 0
+#pragma warning(disable: 4723)
 
 
 void testcaseException_Recursion( unsigned Depth )
-- 
2.41.0.windows.1

